x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/operator/OperatorUserDescriptor.java

```diff
@@ -0,0 +1,274 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+package org.elasticsearch.xpack.security.operator;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.elasticsearch.ElasticsearchException;
+import org.elasticsearch.ElasticsearchParseException;
+import org.elasticsearch.common.ParseField;
+import org.elasticsearch.common.xcontent.DeprecationHandler;
+import org.elasticsearch.common.xcontent.NamedXContentRegistry;
+import org.elasticsearch.common.xcontent.XContentParser;
+import org.elasticsearch.common.xcontent.XContentParserUtils;
+import org.elasticsearch.common.xcontent.XContentType;
+import org.elasticsearch.env.Environment;
+import org.elasticsearch.watcher.FileChangesListener;
+import org.elasticsearch.watcher.FileWatcher;
+import org.elasticsearch.watcher.ResourceWatcherService;
+import org.elasticsearch.xpack.core.XPackPlugin;
+import org.elasticsearch.xpack.core.security.authc.Authentication;
+import org.elasticsearch.xpack.core.security.authc.AuthenticationField;
+import org.elasticsearch.xpack.core.security.authc.file.FileRealmSettings;
+import org.elasticsearch.xpack.core.security.user.User;
+import org.elasticsearch.xpack.core.security.xcontent.XContentUtils;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.StandardOpenOption;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Locale;
+import java.util.Objects;
+import java.util.Set;
+
+public class OperatorUserDescriptor {
+    private static final Logger logger = LogManager.getLogger(OperatorUserDescriptor.class);
+
+    private final Path file;
+    private volatile List<Group> groups;
+
+    public OperatorUserDescriptor(Environment env, ResourceWatcherService watcherService) {
+        this.file =  XPackPlugin.resolveConfigFile(env, "operator_users.yml");
+        this.groups = parseFileLenient(file, logger);
```

Comment:
Updated accordingly

Reply:
Sorry I forgot to mention that the updated code still allows the file to be missing because I think it could be a valid use case. Because operator users are node specific, it might be useful to enable this feature, but only allow operator users to connect on specific nodes for performing the actions. In this case, a missing file would come in handy to block certain nodes from being used to issue operator-only requests. The same thing can be achieved by having an empty list for `usernames` in the config file. But it could provide convenience to allow the file to be simply deleted. Please let me know if you think this is still too much lenience and would prefer to not have it.